### PR TITLE
Move RDF + JS rules to their own folders

### PR DESCRIPTION
**Pull request by tofumatt**
Related to #78; this moves the RDF rules into their own folder and the rule tests into a separate file.

Note: This PR also moves JS rules to their own folder, but in its own commit.

### CODE REVIEW

**Comment by tofumatt** [comment]
Good to r? now

File: src/validators/javascript.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ export default class JavaScriptScanner {
       // pass it the entire source file as a string.
       let eslint = new ESLint.CLIEngine({
         ignore: false,
-        rulePaths: [path.join('src', 'rules')],
+        rulePaths: [path.join('src', 'rules', 'javascript')],
```
Comment (muffinresearch) [inline src/validators/javascript.js]: Is the join needed? Won't this just be 'src/rules/javascript'

File: src/rules/rdf/mustNotExist.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    validatorMessages = validatorMessages.concat(
+      _checkForTags(xmlDoc, namespace, RDF_OBSOLETE_TAGS, VALIDATION_WARNING,
+                    'TAG_OBSOLETE_', filename));
+
+    resolve(validatorMessages);
+  });
+};
+
+export var _checkForTags = (
+(xmlDoc, namespace, tags, severity, errorCodePrefix, filename=null) => {
```
Comment (muffinresearch) [inline src/rules/rdf/mustNotExist.js]: Seems like a weird indent.

Comment (muffinresearch) [inline src/rules/rdf/mustNotExist.js]: Do you need the fat arrow function? 

Also perhaps use an object for the args - that will make it clearer in the caller what args you're passing.

e.g. 

```
function _checkForTags({xmlDoc, namespace, tags, severity, errorCodePrefix, filename=null} = {}) {
  // use any arg passed in the object directly here.
}

_checkForTags({severity: 'hai'});

```

Here's an example showing that sort of thing in action:

http://jsbin.com/cowaki/edit?js,console

**Comment by muffinresearch** [comment]
Looks good to me aside from possibly tweaking the few things in the comments.

r+wc

File: src/rules/rdf/mustNotExist.js (hunk 2 above)
Comment (tofumatt) [inline src/rules/rdf/mustNotExist.js]: Ah, that's a good idea. I too thought the function signature was too long.

File: src/validators/javascript.js (hunk 1 above)
Comment (tofumatt) [inline src/validators/javascript.js]: I always worry about Windows users, in case a module (like `ESLint.CLIEngine`) isn't normalising the paths itself.

**Pull request merged**